emData/generate_MC.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+
+# This script generates MatchCalculator_parameters.h,
+# MatchCalculatorTop.h, and MatchCalculatorTop.cc in the
+# TrackAlgorithm/ directory. Currently supports all MCs for L3_PHIC.
+
+from __future__ import absolute_import
+from __future__ import print_function
+import os,sys, re
+
+TF_index = ['L1L2', 'L2L3', 'L3L4', 'L5L6', 'D1D2', 'D3D4', 'L1D1', 'L2D1']
+TF_index = {k:v for v,k in enumerate(TF_index)}
+
+def ASRegion(region):
+  if region in ['L1', 'L2', 'L3']: return 'BARRELPS'
+  elif region in ['L4', 'L5', 'L6']: return 'BARREL2S'
+  else: return 'DISK'
+
+def APRegion(region):
+  if region in ['L1', 'L2', 'L3']: return 'BARRELPS'
+  elif region in ['L4', 'L5', 'L6']: return 'BARREL2S'
+  else: return 'DISK'
+
+def FMRegion(region):
+  if region in ['L1', 'L2', 'L3', 'L4', 'L5', 'L6']: return 'BARREL'
+  else: return 'DISK'
+
+if len(sys.argv) < 2:
+    print("Usage: " + sys.argv[0] + " WIRES_FILE")
+    sys.exit(1)
+wiresFileName = sys.argv[1]
+
+# First, parse the wires file and store the memory names associated with MCs in
+# dictionaries with the MC names as keys.
+wiresFile = open(wiresFileName)
+CMMems = {}
+FMMems = {}
+for line in wiresFile:
+    # Only barrel-only seeds are supported right now.
+    # Works for MC L3 phi sector C
+    if not any(["MC_" + l + "PHIB" in line for l in ["L3", "L4", "L5", "L6"]]) and \
+       not any(["MC_" + l + "PHIC" in line for l in ["L3"]]): #MC_L3PHIC was used for testing
+        continue
+    line = line.rstrip()
+    tcName = re.sub(r".*MC_(......).*", r"MC_\1", line)
```

Comment:
@bryates Please change `tcName` → `mcName`.

Reply:
Done. I'll push the change soon.